### DISCUSSION TITLE
Feed rates

### DISCUSSION THREAD

**Discussion by Shmiii** [winder/Universal-G-Code-Sender#1822] [Help] [answered]
I’ve just update from 2.0.7. to 2.0.9. I’m having a problem that I didn’t have before where the feed rates seem to be random when I use a macro or any pre programed functions like homing. For example the homing feed rate is setup at 1000mm/min. When I manually key $HY (X or Z) it uses 1000 but if I use the home function all three axis run at 1414. Another example; I have a macro that I use to move the spindle out of the way for loading material, when I use that it’s either way too slow at 500 or way too fast at 4854 or 3104 or whatever seemingly random number it wants to use. The max rates are set to  3100 for both x and y, 500 for z. When jogging it uses the feed rate set in the jog controller. I’ve already checked that the feed speed scaler in the options was unchecked. Any idea what I’m missing?

**Comment by breiler** [comment, marked as answer]
Homing will not be done using the total speed of 1000mm/min, instead each axis is moving at the max speed. So when homing diagonally it will move X and Y in the speed of 1000 mm/min, the hypotenuse of that triangle this creates will be 1414.

Regarding your macros, it depends on what command you are using for movement. If you are using `G0` it will move as fast as the machine will allow (also called rapid movement).

 If you are using G1 it will move using the feed rate set with `F`. Ex:
`G1 X100 Y100 F1000`

If you don't set the feed rate in your macro, it will use the one set from previous commands which could seem a bit random.

**Reply by Shmiii** [reply]
Well that just makes good sense. I knew it was something I was misunderstanding. I’ll be sure to include feed rates in my macros. Thank you for the quick and detailed reply.
